server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -245,7 +245,7 @@ protected void doStart(ClusterState clusterState) {
                             masterNode,
                             actionName,
                             request,
-                            new ActionListenerResponseHandler<Response>(listener, responseReader) {
+                            new ActionListenerResponseHandler<>(listener, responseReader) {
```

Comment:
Could just pass it here I think?

```suggestion
                            new ActionListenerResponseHandler<>(listener, responseReader, executor) {
```

Reply:
:facepalm: fixed